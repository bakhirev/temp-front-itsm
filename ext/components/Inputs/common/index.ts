export * from './interface';
export * from './hooks';
export * from './utils';
export * from './helpers';
export * from './types';
