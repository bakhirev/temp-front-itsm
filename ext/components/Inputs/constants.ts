import { TYPOGRAPHY } from '../Typography';

export const BORDER_TYPE = 'solid';
export const ICON_SIZE_MICRO = 20;
export const ICON_SIZE_DEFAULT = 24;
export const TEXT_MARGIN = 8;
export const ICON_PADDING_RIGHT = 4;
export const INPUT_WIDTH = '100%';
export const INPUT_PADDING = 16;
export const ICON_WRAPPER_PADDING_RIGHT = 15;
export const DEFAULT_WIDTH = '100%';
export const INPUT_SIZE_BIG = 56;
export const INPUT_SIZE_SMALL = 40;
export const INPUT_SIZE_MICRO = 32;
export const USERICON_PADDING_TOP = 5;
export const SUFFIX_PADDING_BIG = 19;
export const SUFFIX_PADDING_SMALL = 11;
export const SUFFIX_PADDING_MICRO = 8;
export const FONT_SIZE_BIG = `${TYPOGRAPHY.fontSize[16]} ${TYPOGRAPHY.fontFamily}`;
export const FONT_SIZE_SMALL = `${TYPOGRAPHY.fontSize[14]} ${TYPOGRAPHY.fontFamily}`;
export const COINS_PADDING_BIG = 17;
export const COINS_PADDING_SMALL = 9;
export const COINS_PADDING_MICRO = 6;
export const COINS_WIDTH_DEFAULT = 22;
export const COINS_WIDTH_MICRO = 20;
export const INCREMENT = 'increment';
export const DECREMENT = 'decrement';
export const DEFAULT_MAX_INPUT_LENGTH = 16;
export const BORDER_BOTTOM_HEIGHT = 2;
export const ICON_PADDING_INPUT_NUMBER_SMALL = 42;
export const ICON_PADDING_INPUT_NUMBER_DEFAULT = 52;
export const ICON_PADDING = 3;
