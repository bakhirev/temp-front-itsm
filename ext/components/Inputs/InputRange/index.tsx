import React, { FC } from 'react';

import { Range, IRangeProps } from '../../Range';
import { IInputDefaultProps } from '../common';

export interface IInputRangeprops {
  /** Добавление суффикса в поле инпута */
  suffix?: string;
  /** Дефолтное значение  */
  defaultValue?: any;
}

export const InputRange: FC<IRangeProps & IInputDefaultProps & IInputRangeprops> = ({
  ...props
}) => {
  return <Range inInput {...props} />;
};
