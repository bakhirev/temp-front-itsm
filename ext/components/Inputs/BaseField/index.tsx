import React, { useState, Dispatch, SetStateAction, ReactNode, FC } from 'react';

import { createTooltipHOC } from '../../TooltipHOC';
import { Body2Short } from '../../Typography';
import { IInputDefaultProps, useTooltipHook } from '../common';
import type { Type } from '../common/types';
import { DEFAULT_WIDTH } from '../constants';

import { InputSkeleton } from './InputSkeleton';
import { Label, Wrapper, Separator, StyledAdditionalText } from './StyledComponents';

export interface IInputProps extends IInputDefaultProps {
  /** Коллбэк на клик */
  onClick?: (event: any) => void | undefined;
  /** иконки из Figma , которые прокидываются в инпут */
  iconServices?: ReactNode;
  /** Отображение подсказки при переполнении инпута. По умолчанию false */
  withTooltip?: boolean;
  /** Тип инпута */
  type?: Type;
  /** Активна ли иконка в инпуте */
  iconIsActive?: boolean;
  /** Функция для переключения active icon */
  setIconIsActive?: Dispatch<SetStateAction<boolean>>;
  /** Использование числового инпута */
  number?: boolean;
  /** Использование Input Range */
  range?: boolean;
  /** Активен ли информер в инпуте */
  activeInformer?: boolean;
  /** Значение инпута */
  value?: any;
  /** Флаг ввода дробной части */
  withCoins?: boolean;
  phoneCountry?: boolean;
  countryIcon?: any;
  selectIcon?: any;
  onCountrySelectOpen?: Function;
  countrySelectComponent?: any;
  onCountrySelect?: Function;
  /** Дефолтное value */
  defaultValue?: any;
}

export const Input: FC<IInputProps> = ({
  disabled,
  label,
  additionalText,
  status = 'default',
  className,
  dataTestId,
  withTooltip = false,
  value = '',
  width = DEFAULT_WIDTH,
  refInput,
  tooltipContainer,
  ...props
}) => {
  const [focused, setFocused] = useState(false);
  const { tooltip, ref } = useTooltipHook(refInput, withTooltip, value);

  const Input = tooltip && !focused ? createTooltipHOC(InputSkeleton) : InputSkeleton;
  const inputElement = (
    <Input
      tooltip={value}
      container={tooltipContainer}
      display="block"
      disabled={disabled}
      focused={focused}
      setFocused={setFocused}
      status={status}
      value={value}
      ref={ref}
      width={width}
      {...props}
    />
  );

  return (
    <Wrapper
      disabled={disabled}
      focused={focused}
      className={className}
      data-test-id={dataTestId}
      width={width}
    >
      <Label>
        {label && (
          <>
            <Body2Short>{label}</Body2Short>
            <Separator />
          </>
        )}
        {inputElement}
      </Label>
      {additionalText && (
        <StyledAdditionalText disabled={disabled} status={status} focused={focused}>
          <Body2Short>{additionalText}</Body2Short>
        </StyledAdditionalText>
      )}
    </Wrapper>
  );
};
