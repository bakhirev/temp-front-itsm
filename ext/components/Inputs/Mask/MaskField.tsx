import React, {
  useRef,
  useEffect,
  useState,
  FocusEvent,
  Dispatch,
  FC,
  ReactNode,
  MutableRefObject,
  ChangeEvent,
  useCallback,
} from 'react';

import { StyledErrorSolid } from '../Text/StyledErrorSolid';
import { Body2Short } from '../../Typography';
import type { Status, Size } from '../common';
import { KEY_CODES } from '../../common';

import { ICON_SIZE_DEFAULT, ICON_SIZE_MICRO } from './constants';
import { getCodePhoneLength, getPhoneNumber, checkPhoneMask, checkMaskFilled } from './utils';
import { PlaceholderMask } from './PlaceholderMask';
import {
  formatValueInMaskValue,
  unFormatValue,
  formatValueOnComplete,
  getInputValue,
} from './MaskUtils';
import {
  Label,
  LabelText,
  StyledInput,
  AdditionalText,
  IconWrapper,
  InputWrapper,
  InputBox,
} from './StyledComponents';

interface IInputMaskFieldProps {
  /** Имя класса для переопределения стилей */
  className?: string;
  /** Лейбел инпута */
  label?: string;
  /** Подсказка в поле ввода */
  placeholder: string;
  /** Коллбек на изменение  значения инпута */
  onChange: any;
  /** Маска инпута */
  mask: string;
  /** Значение инпута */
  value: any;
  /** Коллбек на изменение фокуса */
  onFocus?: (event: FocusEvent<HTMLInputElement>) => void;
  /** Коллбек на изменение  блюр */
  onBlur?: (event: FocusEvent<HTMLInputElement>) => void;
  /** Обработчи для переключения флага focused */
  setFocused: Dispatch<React.SetStateAction<boolean>>;
  /** Флаг  фокусе инпут или нет */
  focused: boolean;
  /** Отключение поля ввода */
  disabled?: boolean;
  /** Статус поля ввода */
  status?: Status;
  /** Ширина инпута */
  width?: string | number;
  /** Дополнительный текст инпута */
  additionalText?: string;
  /** Иконка для инпута */
  icon?: ReactNode;
  /** Размер инпута */
  size?: Size;
  /** Отображать ли иконку ошибки */
  errorIcon?: boolean;
  /** Внешний реф */
  refInput?: any;
  /** Колбек возвращает value когда закончен ввод по маске */
  onComplete?: (value: any) => void;
  /** Колбек для форматирования value */
  formatValue?: (value: Date | Date[] | string) => string;
  /** Колбек для деформатирования value */
  removeFormatValue?: (value: string, value2?: string) => Date | Array<Date | null> | string | null;
  /** Показывать всегда маску */
  alwaysShowMask?: boolean;
  /** Уникальный идентификатор */
  id?: string;
}

export const MaskField: FC<IInputMaskFieldProps> = ({
  className,
  label,
  placeholder,
  onChange,
  onFocus,
  onBlur,
  setFocused,
  alwaysShowMask,
  focused,
  disabled,
  status = 'default',
  additionalText,
  size = 'big',
  mask,
  icon,
  errorIcon,
  refInput,
  value = '',
  onComplete,
  formatValue,
  id,
  removeFormatValue,
}) => {
  const [prevInput, setPrevInput] = useState({ value: '', cursorStart: 0 });
  const [errorActive, setErrorActive] = useState(true);
  const [keyDown, setKeyDown] = useState(false);

  const inputRef: MutableRefObject<HTMLInputElement | null> = useRef(null);

  const inputValue: any = getInputValue(value, mask, focused, formatValue);

  const handleComplete = useCallback(() => {
    onComplete && onComplete(formatValueOnComplete(inputValue, mask, removeFormatValue));
  }, [inputValue, mask, onComplete]);

  useEffect(() => {
    if (checkMaskFilled(mask, inputValue)) {
      handleComplete();
    }
    if (!refInput) {
      if (inputRef.current && prevInput.value !== inputValue) {
        const position = prevInput.cursorStart;
        inputRef.current.setSelectionRange(position, position);
      }
    } else {
      if (refInput.current && prevInput.value !== inputValue) {
        if (inputValue.length === mask.length && !keyDown) {
          refInput.current.setSelectionRange(inputValue.length, inputValue.length);
        } else {
          const position = prevInput.cursorStart;
          refInput.current.setSelectionRange(position, position);
        }
      }
    }
  }, [handleComplete, inputValue, prevInput.cursorStart, prevInput.value]);

  const handleClick = (event: any) => {
    const code = getCodePhoneLength(mask);
    if (event.target.selectionStart < code) {
      event.target.setSelectionRange(code, code);
    }
  };

  const handleKeyUp = (event: any) => {
    const code = getCodePhoneLength(mask);

    if (checkPhoneMask(mask) && event.target.selectionStart < code) {
      event.target.setSelectionRange(code, code);
    }
  };

  const handleKeyDown = (event: any) => {
    const code = getCodePhoneLength(mask);
    setKeyDown(true);
    if (event.keyCode === KEY_CODES.ENTER) {
      inputRef.current?.blur();
      refInput?.current?.blur();
    }
    if (checkPhoneMask(mask) && event.target.selectionStart < code) {
      event.target.setSelectionRange(code, code);
      if (event.target.selectionStart <= code && event.keyCode === KEY_CODES.BACKSPACE) {
        onChange({ value: '', masked: '' });
      }
    }
  };

  const handleFocus = (event: FocusEvent<HTMLInputElement>) => {
    onFocus?.(event);
    setErrorActive(false);
    setFocused(true);
    if (checkPhoneMask(mask)) {
      handleChange(event);
    }
  };

  const handleBlur = (event: FocusEvent<HTMLInputElement>) => {
    onBlur?.(event);
    setFocused(false);
    setKeyDown(false);
    setErrorActive(true);
    if (checkPhoneMask(mask) && value.length <= getPhoneNumber(mask).length) {
      onChange({ value: '', masked: '' });
    }
  };

  const handleChange = (event: ChangeEvent<HTMLInputElement>) => {
    const inputRef = event.target;
    let inputValue;
    if (checkPhoneMask(mask) && value === '') {
      inputValue = getPhoneNumber(mask);
    } else {
      inputValue = inputRef.value;
    }

    const cursorStart = formatValueInMaskValue(
      unFormatValue(inputValue.slice(0, inputRef.selectionStart), mask),
      mask
    ).length;

    setPrevInput({ value: inputValue, cursorStart });

    const newValue = unFormatValue(inputValue, mask);

    if (value !== newValue) {
      onChange({
        value: newValue,
        masked: formatValueInMaskValue(newValue, mask),
      });
    }
  };

  const iconSize = 'micro' === size ? ICON_SIZE_MICRO : ICON_SIZE_DEFAULT;
  const isPlaceholder = !focused && inputValue?.length === 0 ? placeholder : undefined;

  return (
    <InputWrapper>
      <Label className={className}>
        {label && (
          <LabelText>
            <Body2Short>{label}</Body2Short>
          </LabelText>
        )}
        <InputBox focused={focused} disabled={disabled} size={size} status={status}>
          <PlaceholderMask
            alwaysShowMask={alwaysShowMask}
            visibleValue={inputValue}
            mask={mask}
            size={size}
            focused={focused}
            disabled={disabled}
          />

          <StyledInput
            $size={size}
            disabled={disabled}
            id={id ? id : 'input-mask'}
            focused={focused}
            placeholder={isPlaceholder}
            ref={refInput === undefined ? inputRef : refInput}
            onChange={handleChange}
            value={inputValue}
            onKeyUp={handleKeyUp}
            onKeyDown={handleKeyDown}
            onFocus={handleFocus}
            onBlur={handleBlur}
            onClick={handleClick}
            autoComplete="off"
          />
          <IconWrapper>
            {icon && !errorIcon && icon}
            {status === 'error' && errorIcon && !disabled && (
              <StyledErrorSolid width={iconSize} $active={!errorActive} />
            )}
          </IconWrapper>
        </InputBox>
        {additionalText && !disabled && (
          <AdditionalText disabled={disabled} status={status} focused={focused}>
            <Body2Short>{additionalText}</Body2Short>
          </AdditionalText>
        )}
      </Label>
    </InputWrapper>
  );
};
