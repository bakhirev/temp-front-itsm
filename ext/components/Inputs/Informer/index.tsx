import React, { useState, ReactNode, FC } from 'react';

import { ReactComponent as HelpSolid } from '../../Icons/service/HelpSolid.svg';
import { createHintHOC } from '../../HintHOC';
import { Input } from '../BaseField';
import { getIconSize } from '../common';
import type { IInputDefaultProps } from '../common/interface';

import { StyledQuestionSolidIcon } from './StyledQuestionSolidIcon';

export interface IInputInformer extends IInputDefaultProps {
  /** Значение инпута */
  value?: string;
  /** Текст информера */
  message?: string;
  /** Дополнительная иконка */
  icon?: ReactNode;
}

export const InputInformer: FC<IInputInformer> = ({
  value = '',
  disabled,
  size = 'big',
  message = '',
  ...props
}) => {
  const [visible, setVisible] = useState(false);

  const iconSize = getIconSize(size);
  const IconWrapper = ({ onClick }: { onClick: () => void }): JSX.Element => (
    <StyledQuestionSolidIcon active={visible} disabled={disabled} onMouseDown={onClick}>
      <HelpSolid width={iconSize} height={iconSize} />
    </StyledQuestionSolidIcon>
  );

  const ButtonWithHint = createHintHOC(IconWrapper);
  const handleRequestHide = () => {
    !disabled && setVisible(false);
  };

  const handleClick = () => {
    !disabled && setVisible(true);
  };

  const iconServices = (
    <ButtonWithHint
      message={message}
      offset={16}
      size={'small'}
      visible={!disabled ? visible : false}
      onClick={handleClick}
      onRequestHide={handleRequestHide}
    />
  );

  return (
    <Input
      size={size}
      disabled={disabled}
      iconServices={iconServices}
      activeInformer={visible}
      value={value}
      {...props}
    />
  );
};
