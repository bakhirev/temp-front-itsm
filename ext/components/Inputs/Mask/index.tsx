import React, { useState, ReactNode, FocusEvent, FC } from 'react';

import type { Status, Size } from '../common';

import { DEFAULT_MASK, DEFAULT_WIDTH } from './constants';
import { MaskField } from './MaskField';
import { Wrapper } from './StyledComponents';

export interface IInputMaskProps {
  /** Подсказка в поле ввода */
  placeholder?: string;
  /** Значение инпута */
  value: any;
  /** Маска инпута */
  mask?: string;
  /** Коллбек на изменение фокуса */
  onFocus?: (event: FocusEvent<HTMLInputElement>) => void;
  /** Коллбек на изменение  блюр */
  onBlur?: (event: FocusEvent<HTMLInputElement>) => void;
  /** Коллбек на изменение  значения инпута */
  onChange: any;
  /** Имя класса для переопределения стилей */
  className?: string;
  /** data-test-id атрибут для тестирования компонента */
  dataTestId?: string;
  /** Отключение поля ввода */
  disabled?: boolean;
  /** Статус поля ввода */
  status?: Status;
  /** Ширина инпута */
  width?: string | number;
  /** Дополнительный текст инпута */
  additionalText?: string;
  /** Размер инпута */
  size?: Size;
  /** Лейбел инпута */
  label?: string;
  /** Иконка для инпута */
  icon?: ReactNode;
  /** Отображать ли иконку ошибки */
  errorIcon?: boolean;
  /** Внешний реф */
  refInput?: any;
  /** Колбек возвращает value когда закончен ввод по маске */
  onComplete?: (value: any) => void;
  /** Колбек для форматирования value */
  formatValue?: (value: Date | Date[] | string) => string;
  /** Колбек для деформатирования value */
  removeFormatValue?: (value: string, value2?: string) => Date | Array<Date | null> | string | null;
  /** Показывать всегда маску */
  alwaysShowMask?: boolean;
  /** Уникальный идентификатор */
  id?: string;
}

export const InputMask: FC<IInputMaskProps> = ({
  placeholder = '',
  value = '',
  mask = DEFAULT_MASK,
  onChange,
  disabled,
  dataTestId,
  className,
  width = DEFAULT_WIDTH,
  refInput,
  ...props
}: IInputMaskProps) => {
  const [focused, setFocused] = useState(false);

  return (
    <Wrapper
      disabled={disabled}
      width={width}
      focused={focused}
      className={className}
      data-test-id={dataTestId}
    >
      <MaskField
        placeholder={placeholder}
        value={value}
        mask={mask}
        width={width}
        setFocused={setFocused}
        onChange={onChange}
        focused={focused}
        refInput={refInput}
        disabled={disabled}
        {...props}
      />
    </Wrapper>
  );
};
