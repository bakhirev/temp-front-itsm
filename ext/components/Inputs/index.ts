import { TimePicker } from '../TimePicker';

import { InputText } from './Text';
import { InputPassword } from './Password';
import { InputSuggest } from './Suggest';
import { InputInformer } from './Informer';
import { InputNumber } from './Number';
import { InputMask } from './Mask';
import { InputRange } from './InputRange';
import { InputPhoneCountry } from './PhoneCountry';
export { IInputProps } from './BaseField';
export { IInputMaskProps } from './Mask';
export type { IOptionSuggest } from './Suggest';

export const Input = {
  Text: InputText,
  Password: InputPassword,
  Number: InputNumber,
  Mask: InputMask,
  Suggest: InputSuggest,
  Range: InputRange,
  Informer: InputInformer,
  PhoneCountry: InputPhoneCountry,
  Time: TimePicker,
};
