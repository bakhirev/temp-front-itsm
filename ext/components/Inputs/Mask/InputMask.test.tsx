import React from 'react';
import { shallow } from 'enzyme';

import { Input } from '../../Inputs';

import { MaskField } from './MaskField';

jest.mock('../../common', () => ({
  DEFAULT_THEME: {
    color: {
      neutral: {
        50: '',
      },
    },
  },
  Z_INDEXES: {
    INFORMER: 0,
  },
}));

const textError = `Ошибка`;
const label = 'Введите номер телефона';
const changeUsernameSpy = jest.fn();

describe('Input.Mask', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should call onChange when user presses keys in input mask field, with mask 9999 9999 9999 9999 ', () => {
    const wrapper = shallow(
      <Input.Mask
        mask={'9999 9999 9999 9999'}
        value={''}
        onChange={changeUsernameSpy}
        label={label}
        size={'small'}
      />
    );
    const input = wrapper.dive().find(MaskField).dive().find('#input-mask');
    input.simulate('change', { target: { value: '99999999' } });
    expect(changeUsernameSpy).toBeCalledTimes(1);
    expect(changeUsernameSpy).toBeCalledWith({ masked: '9999 9999', value: '99999999' });
  });

  it('should call onChange when user presses keys in input mask field, with mask 99.99.9999', () => {
    const wrapper = shallow(
      <Input.Mask
        mask={'99.99.9999'}
        value={''}
        onChange={changeUsernameSpy}
        label={label}
        size={'small'}
      />
    );

    const input = wrapper.dive().find(MaskField).dive().find('#input-mask');
    input.simulate('change', { target: { value: '99999999' } });
    expect(changeUsernameSpy).toBeCalledTimes(1);
    expect(changeUsernameSpy).toBeCalledWith({ masked: '99.99.9999', value: '99999999' });
  });

  it('should call onChange when user presses keys in input mask field, with mask +7(999) 999 9999', () => {
    const wrapper = shallow(
      <Input.Mask
        mask={'+7(999) 999 9999'}
        value={'343'}
        onChange={changeUsernameSpy}
        label={label}
        size={'small'}
      />
    );

    const input = wrapper.dive().find(MaskField).dive().find('#input-mask');
    input.simulate('change', { target: { value: '99999999999' } });
    expect(changeUsernameSpy).toBeCalledTimes(1);
    expect(changeUsernameSpy).toBeCalledWith({ masked: '+7(999) 999 9999', value: '99999999999' });
  });

  it('should call onChange when user presses keys in input mask field, with mask 99.99.9999 - 99.99.9999', () => {
    const wrapper = shallow(
      <Input.Mask
        mask={'99.99.9999 - 99.99.9999'}
        value={''}
        onChange={changeUsernameSpy}
        label={label}
        size={'small'}
      />
    );

    const input = wrapper.dive().find(MaskField).dive().find('#input-mask');
    input.simulate('change', { target: { value: '9999999999999999' } });
    expect(changeUsernameSpy).toBeCalledTimes(1);
    expect(changeUsernameSpy).toBeCalledWith({
      masked: '99.99.9999 - 99.99.9999',
      value: '9999999999999999',
    });
  });

  it('should render component with error status', () => {
    const wrapper = shallow(
      <Input.Mask
        mask={'+7(999) 999 9999'}
        value={''}
        onChange={(value: any) => value}
        label={label}
        additionalText={textError}
        status={'error'}
        size={'big'}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render component with error status', () => {
    const wrapper = shallow(
      <Input.Mask
        mask={'+7(999) 999 9999'}
        value={''}
        onChange={(value: any) => value}
        label={label}
        additionalText={textError}
        status={'error'}
        size={'big'}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
