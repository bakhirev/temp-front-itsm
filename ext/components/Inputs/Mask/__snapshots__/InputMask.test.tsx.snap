// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input.Mask should render component with error status 1`] = `
<styled.div
  focused={false}
  theme={
    Object {
      "color": Object {
        "neutral": Object {
          "50": "",
        },
      },
    }
  }
  width="100%"
>
  <Component
    additionalText="Ошибка"
    focused={false}
    label="Введите номер телефона"
    mask="+7(999) 999 9999"
    onChange={[Function]}
    placeholder=""
    setFocused={[Function]}
    size="big"
    status="error"
    value=""
    width="100%"
  />
</styled.div>
`;

exports[`Input.Mask should render component with error status 2`] = `
<styled.div
  focused={false}
  theme={
    Object {
      "color": Object {
        "neutral": Object {
          "50": "",
        },
      },
    }
  }
  width="100%"
>
  <Component
    additionalText="Ошибка"
    focused={false}
    label="Введите номер телефона"
    mask="+7(999) 999 9999"
    onChange={[Function]}
    placeholder=""
    setFocused={[Function]}
    size="big"
    status="error"
    value=""
    width="100%"
  />
</styled.div>
`;
